Src/IronPython.Modules/nt.cs

```diff
@@ -351,9 +351,25 @@ public static int dup(CodeContext/*!*/ context, int fd) {
             PythonFileManager fileManager = context.LanguageContext.FileManager;
 
             StreamBox streams = fileManager.GetStreams(fd); // OSError if fd not valid
-            fileManager.EnsureRefStreams(streams);
-            fileManager.AddRefStreams(streams);
-            return fileManager.Add(new(streams));
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) {
```

Comment:
I'm thinking the platform agnostic version should be the else clause. Though it's possible we haven't been very consistent in doing this. For example, something running on Blazor would fall into the else clause (although the `nt` module might not be super relevant in that case).

Reply:
By "the platform agnostic version" do you mean the emulated descriptors version? And that the genuine file descriptors should be limited to Linux and maxOS only? In the past, the platform specific conditions and branches were dictated by the available guard `SupportedOSPlatform`, but now that we have also `UnsupportedOSPlatform` anything is possible. I thought that `UnsupportedOSPlatform` was merely a convenience, but if we were to be considerate of any other platforms than the ones we officially support, it becomes a necessity. OTOH, I see that `IsOSPlatform` accepts only 4 enum values (the 4th being FreeBSD) so how it is supposed to work on Blazor?

Reply:
Yes, I meant using the emulated version when the platform is unknown.

`OSPlatform` can be constructed using arbitrary strings. On .NET it just dispatches to `System.OperatingSystem` which contains all sorts of new goodies. For Blazor the platform string is `BROWSER`. But there are a bunch of others beyond what's in `RuntimeInformation`.